mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,482 @@
+        The metadata contains the evaluation metrics computed on validation
+        data with that checkpoint along with the current epoch. It depends
+        on `EvalHook` whose priority is more than MMDetWandbHook.
+
+    - Evaluation: At every evaluation interval, the `MMDetWandbHook` logs the
+        model prediction as interactive W&B Tables. The number of samples
+        logged is given by `num_eval_images`. Currently, the `MMDetWandbHook`
+        logs the predicted bounding boxes along with the ground truth at every
+        evaluation interval. This depends on the `EvalHook` whose priority is
+        more than `MMDetWandbHook`. Also note that the data is just logged once
+        and subsequent evaluation tables uses reference to the logged data
+        to save memory usage. Please refer to
+        https://docs.wandb.ai/guides/data-vis to learn more about W&B Tables.
+
+    For more details check out W&B's MMDetection docs:
+    https://docs.wandb.ai/guides/integrations/mmdetection
+
+    ```
+    Example:
+        log_config = dict(
+            interval=10,
+            hooks=[
+                dict(type='MMDetWandbHook',
+                     init_kwargs={
+                         'entity': WANDB_ENTITY,
+                         'project': WANDB_PROJECT_NAME
+                     },
+                     interval=50,
+                     log_checkpoint=True,
+                     num_eval_images=100,
+                     bbox_score_thr=0.3)
+            ])
+    ```
+
+    Args:
+        init_kwargs (dict): A dict passed to wandb.init to initialize
+            a W&B run. Please refer to https://docs.wandb.ai/ref/python/init
+            for possible key-value pairs.
+        interval (int): Logging interval (every k iterations).
+            Default 10.
+        log_checkpoint (bool): Save the checkpoint at every checkpoint interval
+            as W&B Artifacts. Use this for model versioning where each version
+            is a checkpoint.
+            Default: False
+        num_eval_images (int): Number of validation images to be logged.
+            Default: 100
+        bbox_score_thr (float): Threshold for bounding box scores.
+    """
+
+    def __init__(self,
+                 init_kwargs=None,
+                 interval=50,
+                 log_checkpoint=False,
+                 num_eval_images=100,
+                 bbox_score_thr=0.3,
+                 **kwargs):
+        super(MMDetWandbHook, self).__init__(init_kwargs, interval, **kwargs)
+
+        self.log_checkpoint = log_checkpoint
+        self.num_eval_images = num_eval_images
+        self.bbox_score_thr = bbox_score_thr
+        self.log_evaluation = True
+        self.best_score = 0
+
+    @master_only
+    def before_run(self, runner):
+        super(MMDetWandbHook, self).before_run(runner)
+
+        # Load config.json file from work_dir
+        load_config_path = osp.join(runner.work_dir, 'config.py')
```

Comment:
```python
        src_cfg_path = osp.join(runner.work_dir, runner.meta['exp_name'])
        if osp.exists(src_cfg_path):
               self.wandb.save(src_cfg_path)
```

Reply:
Unfortunately for `runner.meta['exp_name']` I am getting `TypeError: 'NoneType' object is not subscriptable`.

Reply:
Hey, it's impossible, I've already run it without a problem.

Reply:
Please confirm it, thank you.